### PR TITLE
feat(functions): Add task queue API support

### PR DESCRIPTION
**Pull request by jonathanedey**
Includes implementation of:
- Functions Service
- taskqueue
- enqueue
- delete
- TaskOptions

### CODE REVIEW

File: firebase_admin/functions.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
+# Copyright 2023 Google Inc.
```
Comment (lahirumaramba) [inline firebase_admin/functions.py:1]: nit: 2024

File: firebase_admin/functions.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+
+_CLOUD_TASKS_API_RESOURCE_PATH = \
+    'projects/{project_id}/locations/{location_id}/queues/{resource_id}/tasks'
+_CLOUD_TASKS_API_URL_FORMAT = \
+    'https://cloudtasks.googleapis.com/v2/' + _CLOUD_TASKS_API_RESOURCE_PATH
+_FIREBASE_FUNCTION_URL_FORMAT = \
+    'https://{location_id}-{project_id}.cloudfunctions.net/{resource_id}'
+
+_FUNCTIONS_HEADERS = {
```
Comment (lahirumaramba) [inline firebase_admin/functions.py:50]: I think we can move some of these constants out of the global scope and into a mode limited scope. For example, can we defined `_FUNCTIONS_HEADERS` inside the `TaskQueue` class?

File: firebase_admin/functions.py [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+
+    Raises:
+        ValueError: If the input arguments are invalid.
+    """
+    return _get_functions_service(app).task_queue(function_name, extension_id)
+
+class _FunctionsService:
+    """Service class that implements Firebase Functions functionality."""
+    def __init__(self, app: App):
+        self._project_id = app.project_id
```
Comment (lahirumaramba) [inline firebase_admin/functions.py:96]: let's validate the project id here. For example:
```
if not project_id:
            raise ValueError(
                'Project ID is required to access Cloud Messaging service. Either set the '
                'projectId option, or use service account credentials. Alternatively, set the '
                'GOOGLE_CLOUD_PROJECT environment variable.')
```

File: firebase_admin/functions.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+                headers=_FUNCTIONS_HEADERS,
+                json={'task': task_payload.__dict__}
+            )
+            task_name = resp.get('name', None)
+            task_resource = \
+                self._parse_resource_name(task_name, f'queues/{self._resource.resource_id}/tasks')
+            return task_resource.resource_id
+        except requests.exceptions.RequestException as error:
+            # TODO: Error handle
+            raise FirebaseError(400, 'Error', http_response=error.response)
```
Comment (lahirumaramba) [inline firebase_admin/functions.py:170]: We might be able to use https://github.com/firebase/firebase-admin-python/blob/23765843a124065dcb8fd5a86a3d8f3b784df34c/firebase_admin/_utils.py#L101 here to handle the GCP errors with the correct error codes

**Review by lahirumaramba — COMMENTED** [review]
Thanks! Looks great overall! Added a few comments.
Let's make sure we have enough test coverage for resource name parsing functions

File: firebase_admin/functions.py (hunk 2 above)
Comment (lahirumaramba) [reply firebase_admin/functions.py:50]: Actually, ignore this. I think the module level constants are okay!

**Comment by jonathanedey** [comment]
Thanks for the review! Going to throw the unit and integration tests into the PR after this one.

**Review by lahirumaramba — APPROVED** [review]

**Pull request merged**